e2e/integration/helpers/helpers.go

```diff
@@ -0,0 +1,341 @@
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/kubeedge/kubeedge/e2e/integration/utils"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+)
+
+const (
+	podLabelSelector = "?labelSelector="
+)
+
+func SenHttpRequest(method, reqapi string) (error, *http.Response){
+	var body io.Reader
+	var resp *http.Response
+
+	client := &http.Client{}
+	req, err := http.NewRequest(method, reqapi, body)
+	if err != nil {
+		// handle error
+		utils.Failf("Frame HTTP request failed: %v", err)
+		return err, resp
+	}
+	req.Header.Set("Content-Type", "application/json")
+	t := time.Now()
+	resp, err = client.Do(req)
+	utils.InfoV6("%s %s %v in %v", req.Method, req.URL, resp.Status, time.Now().Sub(t))
+	if err != nil {
+		// handle error
+		utils.Failf("HTTP request is failed :%v", err)
+		return err, resp
+	}
+
+	return nil, resp
+}
+
+func GetDeployments(list *metav1.DeploymentList, getDeploymentApi string) error{
+
+	err, resp := SenHttpRequest(http.MethodGet, getDeploymentApi)
+	defer resp.Body.Close()
+	contents, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		utils.Failf("HTTP Response reading has failed: %v", err)
+		return err
+	}
+	err = json.Unmarshal(contents, &list)
+	if err != nil {
+		utils.Failf("Unmarshal HTTP Response has failed: %v", err)
+		return err
+	}
+	return nil
+
+}
+
+func GenerateDeploymentBody(deploymentName, image string, port, replica int) (error, map[string]interface{}){
+	var temp map[string]interface{}
+
+	//containerName := "app-" + utils.GetRandomString(5)
+	Body := fmt.Sprintf(`{"apiVersion": "apps/v1","kind": "Deployment","metadata": {"name": "%s","labels": {"app": "nginx"}},
+				"spec": {"replicas": %v,"selector": {"matchLabels": {"app": "nginx"}},"template": {"metadata": {"labels": {"app": "nginx"}},
+				"spec": {"containers": [{"name": "nginx","image": "%s"}]}}}}`, deploymentName, replica, image)
+	err:= json.Unmarshal([]byte(Body), &temp)
+	if err!= nil{
+		utils.Failf("Unmarshal body failed: %v", err)
+		return err, temp
+	}
+
+	return nil, temp
+}
+
+//Function to handle app deployment/delete deployment.
+func HandleDeployment(operation string, apiserver string, UID string, ImageUrl string, replica int) bool {
+	var req *http.Request
+	var err error
+	var body io.Reader
+
+	client := &http.Client{}
+	switch operation {
+	case "POST":
+		Port := utils.RandomInt(10, 100)
+		err, body := GenerateDeploymentBody(UID, ImageUrl, Port, replica)
+		if err != nil {
+			utils.Failf("Payload marshalling failed: %v", err)
```

Comment:
"Payload marshalling failed"  -> Is this correct ? Aren't we performing Unmarshal operation  inside GenerateDeploymentBody() ??